PCP/server/src/api/cart_api.py

```diff
@@ -0,0 +1,42 @@
+from flask import jsonify
+from flask_restful import Resource
+
+try:
+    from src.db.cart import get_cart_contents
+except ImportError:
+    from db.cart import get_cart_contents, add_item_to_cart, remove_item_from_cart
+
+
+class Cart(Resource):
+    def get(self, user_id):
+        """Fetches all items in a user's shopping cart by their user ID."""
+        try:
+            cart_items = get_cart_contents(user_id)
+            if not cart_items:
+                return jsonify({"error": "No items found in the cart"}), 404
+            return jsonify({"cart_items": cart_items})
+        except Exception as e:
+            return jsonify({"error": str(e)}), 500
+
+
+class AddToCart(Resource):
+    def post(self, user_id, product_id, quantity):
+        """Adds a new item to the shopping cart or updates the quantity if it already exists."""
+        try:
+            result = add_item_to_cart(user_id, product_id, quantity)
```

Comment:
does it takes session_key as a input !! as I have explained it previously that user can only send the session_key and u guys have to put the code which validates the user based on the session_key and with that it stores the product_id in that user_id !!